PropertyChanged.Fody/WarningChecker.cs

```diff
@@ -66,18 +67,26 @@ public void EmitConditionalWarning(ICustomAttributeProvider member, string messa
     {
         if (SuppressWarnings)
             return;
-        
-        if (member.HasCustomAttributes && member.CustomAttributes.ContainsAttribute("PropertyChanged.SuppressPropertyChangedWarningsAttribute"))
+
+        var suppressAttrName = "PropertyChanged.SuppressPropertyChangedWarningsAttribute";
+
+        if (member.HasCustomAttributes && member.CustomAttributes.ContainsAttribute(suppressAttrName))
             return;
+
+        if (member is IMemberDefinition memberDefinition && memberDefinition.DeclaringType.GetAllCustomAttributes().ContainsAttribute(suppressAttrName))
```

Comment:
This makes `[SuppressPropertyChangedWarnings]` inheritable from a base class. I'm not quite sure that's the best thing to do as it can easily lead to unintended warning suppressions.

I think I'd prefer a more granular way to suppress warnings:

```suggestion
        if (member is IMemberDefinition memberDefinition &&
            memberDefinition.DeclaringType.HasCustomAttributes &&
            memberDefinition.DeclaringType.CustomAttributes.ContainsAttribute(suppressAttrName))
```

Thoughts?

Reply:
Will it suppress warnings if overridden?

Reply:
I'm not sure I understand the question, but my suggestion here changes the behavior: you always need to put the attribute on the *same* class you get the warnings on, placing it on the base class will not suffice.

This is up for discussion. If you'd like to keep the attribute inheritance, could you explain why it would be more useful than dangerous?

Reply:
I committed it, but I think we should disallow class-level inheritance.

I'm not sure about member suppression inheritance, does that mean that we'll need to override all those properties in their superclasses and re-suppress their warnings?

Reply:
> I committed it, but I think we should disallow class-level inheritance.

Then that commit disallows it. :+1:

> I'm not sure about member suppression inheritance, does that mean that we'll need to override all those properties in their superclasses and re-suppress their warnings?

No, the properties will be handled only once, in their base class. There's no need to override them. I was only talking about class-level inheritance here.

Reply:
Then we should be fine without inheritance.